### PR TITLE
[v4] Fix(types): external module type error (CompoundedComponent)

### PR DESCRIPTION
**Pull request by wangcch**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/pull/38666

### 💡 Background and solution

Wrong type for `CompoundedComponent` from external module
For V4.x

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fix 'CompoundedComponent' type error from external module   |
| 🇨🇳 Chinese |  修复外部暴露类 `CompoundedComponent` 的组建的类型报错  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by yoyo837** [comment]
Rebase please.

**Comment by wangcch** [comment]
> Rebase please.

Done

File: components/alert/index.tsx [hunk 1]
```diff
@@ -69,7 +69,7 @@ interface IconNodeProps {
   description: AlertProps['description'];
 }
 
-const IconNode: React.FC<IconNodeProps> = props => {
+const IconNode: React.FC<IconNodeProps> = (props) => {
```
Comment (yoyo837) [inline components/alert/index.tsx:72]: 这些括号的改动能避免吗? 有点多不好review.

Comment (yoyo837) [reply components/alert/index.tsx:72]: 看看你本地prettier 设置

Comment (wangcch) [reply components/alert/index.tsx:72]: 这个之前的 PR 有说明，是 .prettierrc ("arrowParens": "avoid") 和 rome js config 冲突

Comment (wangcch) [reply components/alert/index.tsx:72]: https://github.com/ant-design/ant-design/pull/38722

Comment (wangcch) [reply components/alert/index.tsx:72]: git hook: `rome format` when pre-commit

Comment (yoyo837) [reply components/alert/index.tsx:72]: 把本地git hook 干掉

Comment (wangcch) [reply components/alert/index.tsx]: 可以

Comment (yoyo837) [reply components/alert/index.tsx]: 还有一部分, 能都还原吗?

Comment (wangcch) [reply components/alert/index.tsx]: 剩下的都是必要修改行，随带按照 rome config 修改了
截屏2023-01-03 11 32 42

**Review by yoyo837 — APPROVED** [review]

**Pull request merged**